modules/svg/svg.go

```diff
@@ -41,6 +41,21 @@ func Init() error {
 	return nil
 }
 
+func MockIcon(icon string) func() {
+	if svgIcons == nil {
+		svgIcons = make(map[string]string)
+	}
+	orig, exist := svgIcons[icon]
+	svgIcons[icon] = fmt.Sprintf(`<svg class="svg %s" width="%d" height="%d"></svg>`, icon, defaultSize, defaultSize)
+	return func() {
+		if exist {
+			svgIcons[icon] = orig
+		} else {
+			delete(svgIcons, icon)
+		}
+	}
+}
+
```

Comment:
Would instead add a `gitea-empty` SVG to the repo. I tried that recently but there was a problem in the JS SVG parser because minifier minifies to `<svg/>` and the JS parser can not correctly parse that with the self-closing tag.

Reply:
I don't think so. I think current approach is clear enough, see the test cases, this content is easy to read, won't make the test code messy.

![image](https://github.com/go-gitea/gitea/assets/2114189/092c3bb6-ec5d-48b7-ab45-48423252c06b)

Reply:
It's just that the frontend may also want to eventually render an empty SVG and it would be better to use the shared SVG infrastructure.

Reply:
That's a different problem ....

This PR is just to make the SVG icons mockable, to test the backend code. And I do not see how it could "share SVG infrastructure" with this PR. This PR requires to construct a mocked SVG with a given name.

Reply:
Oh, I thought this was also rendered. Is it purely for testing?

Reply:
> Oh, I thought this was also rendered. Is it purely for testing? If yes, I would move it into a test-only file, so it doesn't even compile into the application.

1. It is not that easy to "move", due to Golang's language limitation
2. It won't compile into the binary since the application doesn't call `MockIcon`

I think nothing needs to worry.

Reply:
Maybe worth introducing a `test_utils.go` at some point. This is usually what I do for test-only utility code.

Reply:
> Maybe worth introducing a `test_utils.go` at some point. This is usually what I do for test-only utility code.

If won't be that easy. Actually there are many "test" related packages, and I have done a lot, for example:  Decouple unit test code from business code #17623 

But for this case:

* If you put `svg_test_utils.go` in `svg` package, then there is no difference as putting in the `svg.go`.
* If you put `svg_test_utils.go` into another package, it couldn't access the private variables in `svg` package.

Anyway I think current approach is good enough.

Reply:
Yes, ok for now.